app/src/main/java/com/android/streetworkapp/MainActivity.kt

```diff
@@ -215,6 +219,15 @@ fun StreetWorkApp(
                 // profile screen + list of friend
                 composable(Screen.PROFILE) {
                   ProfileScreen(navigationActions, userViewModel, innerPadding)
+                  val showSettingsDialog = remember { mutableStateOf(false) }
+
+                  screenParams?.topAppBarManager?.setActionCallback(
+                      TopAppBarManager.TopAppBarAction.SETTINGS) {
+                        showSettingsDialog.value = true
+                      }
+
+                  // The settings "in" the profile screen
+                  SettingDialog(showSettingsDialog)
```

Comment:
I specifically asked @Alex720p for advice on what I had done because he is the one who made the scaffold transition. 

Initially I passed the topAppBarManager as an argument, but we concluded that it goes against the spirit of separating the scaffold with manager in MainActivity (have customizable topAppBars for each screens).